### PR TITLE
Add test_voq_fabric_capacity.py

### PR DESCRIPTION
**Pull request by jfeng-arista**
### Description of PR

Summary: Add test_voq_fabric_capacity.py to test fabric capacity monitoring and related CLI.
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

### CODE REVIEW

**Comment by jfeng-arista** [comment]
@kenneth-arista @arista-nwolfe  @arlakshm

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml [hunk 1]
```diff
@@ -2146,6 +2146,12 @@ voq/test_fabric_reach.py:
     conditions:
       - "('t2' not in topo_name) or (asic_subtype not in ['broadcom-dnx']) or ('arista_7800' not in platform) or (asic_type in ['cisco-8000'])"
 
+voq/test_voq_fabric_capacity.py:
+  skip:
+    reason: "Skip test_voq_fabric_capacity on unsupported testbed."
+    conditions:
+      - "('t2' not in topo_name) or (asic_subtype not in ['broadcom-dnx']) or ('arista_7800' not in platform) or (asic_type in ['cisco-8000'])"
```
Comment (arlakshm) [inline tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:2153]: This tests should be supported on all DNX platforms? please remove the check `('arista_7800' not in platform)`

File: tests/voq/test_voq_fabric_capacity.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    if len(up_link_list) > 0:
+        shutlink = random.choice(up_link_list)
+    else:
+        return
+
+    # Start the test. Isolate a link and check if the capacity command get updated.
+    # Unisolate the link and check if the capacity command get updated.
+    try:
+        if duthost.is_multi_asic:
+            cmdPrefix = "sonic-db-cli -n asic{} APPL_DB hset 'FABRIC_PORT_TABLE:Fabric{}'".format(asic, shutlink)
```
Comment (arlakshm) [inline tests/voq/test_voq_fabric_capacity.py:87]: can we use the CLI `sudo config fabric port isolate` to isolate a fabric port

**Comment by saksarav-nokia** [comment]
@jfeng-arista , just wondering if you need to add a test to verify the log event when the operating links reach the warning threshold limit

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml (hunk 1 above)
Comment (jfeng-arista) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:2153]: > This tests should be supported on all DNX platforms? please remove the check `('arista_7800' not in platform)`

I can remove it . I see all other fabric tests are stil running on arista systems, so added before.

File: tests/voq/test_voq_fabric_capacity.py (hunk 2 above)
Comment (jfeng-arista) [reply tests/voq/test_voq_fabric_capacity.py:87]: > can we use the CLI `sudo config fabric port isolate` to isolate a fabric port

revised

**Comment by jfeng-arista** [comment]
> @jfeng-arista , just wondering if you need to add a test to verify the log event when the operating links reach the warning threshold limit

good point. I can do that in the next review. thank you for the suggestion!

**Comment by jfeng-arista** [comment]
2024-12-12T19:13:48.0517735Z Test summary:
2024-12-12T19:13:48.0518013Z {
2024-12-12T19:13:48.0518303Z     "errors": 1,
2024-12-12T19:13:48.0518566Z     "failures": 4,
2024-12-12T19:13:48.0518817Z     "skipped": 642,
2024-12-12T19:13:48.0519223Z     "tests": 2152,
2024-12-12T19:13:48.0519633Z     "times": 95057.903,
2024-12-12T19:13:48.0520044Z     "xfails": 6,
2024-12-12T19:13:48.0520434Z     "passes": 1499
2024-12-12T19:13:48.0520782Z }
2024-12-12T19:13:48.0521224Z Error type: RUN_TEST_CASE_FAILED
2024-12-12T19:13:48.0522148Z Error message: snmp/test_snmp_memory.py|||2 failed for RUN_TEST_CASE_FAILED, stop whole test plan
2024-12-12T19:13:48.0523451Z Operation failed with exception: Exception('Test plan id: 675b1e9bb678953935e62c3e, status: FAILED, result: FAILED, Elapsed 3971 seconds. Check https://elastictest.org/scheduler/testplan/675b1e9bb678953935e62c3e for test plan status')
2024-12-12T19:13:48.0695872Z All testplan failed, cancel following steps
2024-12-12T19:13:48.0706083Z 
2024-12-12T19:13:48.0731823Z ##[error]Bash exited with code '3'.
2024-12-12T19:13:48.0774079Z ##[section]Finishing: Run test


the failure does not look like relates the change

**Review by arlakshm — COMMENTED** [review]
lgtm

**Comment by arlakshm** [comment]
@saksarav-nokia, can you approve this PR

**Comment by saksarav-nokia** [comment]
@arlakshm , lgtm

**Review by arlakshm — APPROVED** [review]

**Pull request merged**